symengine/tests/basic/test_basic.cpp

```diff
@@ -383,6 +383,13 @@ TEST_CASE("compare: Basic", "[basic]")
     r2 = mul(x, z);
     CHECK(r1->compare(*r2) == -1);
     CHECK(r2->compare(*r1) == 1);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
+
+    r1 = mul(y, x);
+    r2 = mul(x, z);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
```

Comment:
I didn't realize it before, but it is indeed platform dependent. Since the hash is platform dependent, the order of the elements in the `Mul` dictionary is platform dependent, therefore the comparison is platform dependent.